sycl/doc/EnvironmentVariables.md

```diff
@@ -250,6 +250,7 @@ variables in production code.</span>
 | `SYCL_PI_LEVEL_ZERO_SINGLE_ROOT_DEVICE_BUFFER_MIGRATION` | Integer | When set to "0" tells to use single root-device allocation for all devices in a context where all devices have same root. Otherwise performs regular buffer migration. Default is 1. |
 | `SYCL_PI_LEVEL_ZERO_REUSE_DISCARDED_EVENTS` | Integer |  When set to a positive value enables the mode when discarded Level Zero events are reset and reused in scope of the same in-order queue based on the dependency chain between commands. Default is 1. |
 | `SYCL_PI_LEVEL_ZERO_EXPOSE_CSLICE_IN_AFFINITY_PARTITIONING` (Deprecated) | Integer | When set to non-zero value exposes compute slices as sub-sub-devices in `sycl::info::partition_property::partition_by_affinity_domain` partitioning scheme. Default is zero meaning that they are only exposed when partitioning by `sycl::info::partition_property::ext_intel_partition_by_cslice`. This option is introduced for compatibility reasons and is immediately deprecated. New code must not rely on this behavior. Also note that even if sub-sub-device was created using `partition_by_affinity_domain` it would still be reported as created via partitioning by compute slices. |
+| `SYCL_PI_LEVEL_ZERO_IMMEDIATE_COMMANDLISTS_EVENT_CLEANUP_THRESHOLD` | Integer | If a non-negative value then the threshold becomes equal to this value. If a negative value then the threshold is INT_MAX. If number of events in the immediate command list exceeds this threshold then cleanup process for those events is executed. It allows to control recycling of events. When this threshold is low then status of the events is checked more often which may allow to recycle events faster, on the other hand makind it too low may result in unnecessary status checks. Default is 100. |
```

Comment:
```suggestion
| `SYCL_PI_LEVEL_ZERO_IMMEDIATE_COMMANDLISTS_EVENT_CLEANUP_THRESHOLD` | Integer | If a non-negative value then the threshold becomes equal to this value. If a negative value then the threshold is INT_MAX. If number of events in the immediate command list exceeds this threshold then cleanup process for those events is executed. It allows to control recycling of events. When this threshold is low then status of the events is checked more often which may allow to recycle events faster, on the other hand makind it too low may result in unnecessary status checks. Default is 20. |
```

Reply:
oh, sorry, fixed.

Reply:
If non-negative then the threshold is set to this value. If negative, the threshold is set to INT_MAX. Whenever the number of events associated with an immediate command list exceeds this threshold, a check is made for signaled events and these events are recycled. Setting this threshold low causes events to be checked more often, which could result in unneeded events being recycled sooner. However, more frequent event status checks may cost time. The default is 20.

Reply:
Sounds good, thank you!